util/codec/codec.go

```diff
@@ -352,6 +353,232 @@ func encodeHashChunkRowIdx(sc *stmtctx.StatementContext, row chunk.Row, tp *type
 	return
 }
 
+// HashChunkColumns writes the encoded value of each row's column, which of index `colIdx`, to h.
+func HashChunkColumns(sc *stmtctx.StatementContext, h []hash.Hash64, chk *chunk.Chunk, tp *types.FieldType, colIdx int, buf []byte, isNull []bool) (err error) {
+	const (
+		sizeUint64  = unsafe.Sizeof(uint64(0))
+		sizeFloat64 = unsafe.Sizeof(float64(0))
+	)
```

Comment:
We should keep constraints in package level, but not in function so that we do not need to calculate `unsafe.Sizeof` each time we call `HashChunkColumns`.